### DISCUSSION TITLE
FormBuilderInterface ignores all constraints when type is defined

### DISCUSSION THREAD

**Discussion by sovetski** [symfony/symfony#57751] [Q&A]
### Description

When we define the `type` of property in a FormType, we lost all constraints defined in the Entity level.

### Example

This code works normally:

```php
public function buildForm(FormBuilderInterface $builder, array $options): void
{
    $builder
        ->add('myproperty')
    ;
}
```

Entity:
```php
#[Assert\Length(min: 70, max: 155)]
#[ORM\Column(length: 255, nullable: true)]
private ?string $myproperty= null;
```

But when I add the `type` like this:

```php
public function buildForm(FormBuilderInterface $builder, array $options): void
{
    $builder
        ->add('myproperty', TextareaType::class)
    ;
}
```

It ignores all constraints for `myproperty`, I lost everything already added on Entity level, it does not add anymore the custom attributes like `maxlength` etc..I have to duplicate the constraints on FormType. It is very anoying, why it works like this? Is it possible to change this behaviour?

**Comment by MatTheCat** [comment]
See https://symfony.com/doc/current/form/type_guesser.html:

> Guessers are used only in the following cases:
> - […]
> - Calling `add()` […] without an explicit type, in a context where the parent form has defined a data class.

Also, guessers do not guess constraints: in your case the only missing data is the `maxlength` attribute (maybe the `required` one too?). You can always open an issue and see how it goes.

**Reply by sovetski** [reply]
When I remove the explicit type from `add()` method, the constraint work as expected, that is why I created it as an issue but someone transformed it to the discussion

**Reply by xabbuh** [reply]
This is the intended behaviour as form type guessers are only applied when you omit the explicit type.
